### DISCUSSION TITLE
App goes to '/' for split second even when initialRoute is set to 'sign-in'

### DISCUSSION THREAD

**Discussion by kyusungpark** [expo/router#886] [Q&A]
Hey everyone,

I have been trying to figure this issue out, but wasn't able to so I thought I ask here.

I have an app that requires users to sign in, so it will attempt to load sign-in screen before anything else.
I noticed that whenever I load the app, it will attempt to go to '/', then renders '/' and lastly lands on 'sign-in' screen.
I did set the initialRouteName to the sign in component like below:
<img width="543" alt="Screenshot 2023-09-20 at 9 53 48 PM" src="https://github.com/expo/router/assets/58890841/4fe28a50-3952-4d17-be13-775255b0675b">
This is a problem for me because when I build a preview, then click the back arrow(<) on the bottom provided by the app itself, it shows the home screen for split second.

Would love to get some ideas from the community.
Thanks,

**Comment by marklawlor** [comment]
Can you please share your repo, the screenshot is not enough for us to assist you.

**Reply by kyusungpark** [reply]
@marklawlor https://github.com/lit-ai/reindeer-web. here is the link thanks! I did change the code a little thinking that it was an issue with redux, but still the same issue.

**Reply by marklawlor** [reply]
When debugging issue, I recommend trying to replicate the behaviour in the example starter project. That way its both easier for you to debug and for others to understand your intentions. Are you able to create a minimal reproduction of your issue?

From my quick glance, your AuthProvider looks suspicious.   This `useEffect` will fire on every route change https://github.com/lit-ai/reindeer-web/blob/1d3220119df413cd2631df013d71116c0ed46f07/hooks/useProtectedRoute.tsx#L35C1-L37

We recently updated the Expo Router authentication docs, so they may be able to assist you as well. https://docs.expo.dev/router/reference/authentication/

**Reply by Scr3nt** [reply]
But on the new doc you still have a loading screen before being redirect to the right screen 
We need to have the check during the splashscreen and after the splashscreen is gone have the right screen directly like on React Navigation
